website/docs/r/ecs_service.html.markdown

```diff
@@ -63,6 +63,7 @@ into consideration during task placement. The maximum number of
 * `placement_constraints` - (Optional) rules that are taken into consideration during task placement. Maximum number of
 `placement_constraints` is `10`. Defined below.
 * `network_configuration` - (Optional) The network configuration for the service. This parameter is required for task definitions that use the awsvpc network mode to receive their own Elastic Network Interface, and it is not supported for other network modes.
+* `wait_for_steady_state` - (Optional) If `true`, Terraform will wait for the service to reach a steady state (like [`aws ecs wait services-stable`](https://docs.aws.amazon.com/cli/latest/reference/ecs/wait/services-stable.html)) before continuing. Default `false`.
```

Comment:
I wanted to write an acceptance test for when this is `true`, but I couldn't come up with a reliable, repeatable way to test we *weren't* hitting a race condition.

This would also be the first test case that required container instances (or which used Fargate and was locked to `us-east-1`.)

I have verified manually that this waits for a new deployment to finish.

Reply:
What race conditions were you thinking of?

I would have thought that you could simply create an ECS service in one step (using your `wait_for_steady_state` to make sure it doesn't complete until there is a task running somewhere) and then another step that increases the `desired_count` and then another step that changes the task definition by setting an environment variable or something.

Creating container instances shouldn't be difficult, I'd just use the latest ECS optimised AMI and set the user data so that it joins the ECS cluster you create for the acceptance test.

Reply:
If we create an ALB and Target Group then the acceptance test can be something as simple as launching `busybox` `sleep 600` in that service.

Since it will never pass health checks the service will never become stable.